libs/island-ui/contentful/src/lib/richTextRendering.tsx

```diff
@@ -9,33 +9,79 @@ import {
 } from '@contentful/rich-text-types'
 import { Asset } from 'contentful'
 import { documentToReactComponents } from '@contentful/rich-text-react-renderer'
-import Image from './Image/Image'
-import FaqList from './FaqList/FaqList'
-import { Slice } from '@island.is/api/schema'
-import { Statistics } from './Statistics/Statistics'
+import { Image, ImageProps } from './Image/Image'
+import FaqList, { FaqListProps } from './FaqList/FaqList'
+import { Statistics, StatisticsProps } from './Statistics/Statistics'
 import Hyperlink from './Hyperlink/Hyperlink'
-import AssetLink from './AssetLink/AssetLink'
+import { AssetLink, AssetLinkProps } from './AssetLink/AssetLink'
 import {
   Typography,
   Blockquote,
   Box,
   TypographyProps,
   ResponsiveSpace,
 } from '@island.is/island-ui/core'
-import ProcessEntry from './ProcessEntry/ProcessEntry'
-import EmbeddedVideo from './EmbeddedVideo/EmbeddedVideo'
+import { ProcessEntry, ProcessEntryProps } from './ProcessEntry/ProcessEntry'
+import EmbeddedVideo, {
+  EmbeddedVideoProps,
+} from './EmbeddedVideo/EmbeddedVideo'
 import StaticHtml from './StaticHtml/StaticHtml'
 import slugify from '@sindresorhus/slugify'
-import { SectionWithImage } from './SectionWithImage/SectionWithImage'
-import { TeamList } from './TeamList/TeamList'
-import { ContactUs } from './ContactUs/ContactUs'
+import {
+  SectionWithImage,
+  SectionWithImageProps,
+} from './SectionWithImage/SectionWithImage'
+import { TeamList, TeamListProps } from './TeamList/TeamList'
+import { ContactUs, ContactUsProps } from './ContactUs/ContactUs'
+
+type HtmlSlice = { __typename: 'Html'; document: any }
```

Comment:
Is document not of type `import { Document } from '@contentful/rich-text-types'`

Reply:
That is true, but the graphql server (or at least the type generated from the query) returns it as a JSON ({ [key: string]: any }) type, so if the use the Document type here, the types won't match

Reply:
Make sense!